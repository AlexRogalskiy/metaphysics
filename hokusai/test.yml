---
version: '2'
services:
  metaphysics:
    command: yarn test
    extends:
      file: common.yml
      service: metaphysics
    environment:
    - NODE_ENV=test
    env_file: ../.env.test
