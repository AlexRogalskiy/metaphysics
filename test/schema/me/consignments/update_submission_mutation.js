import { runAuthenticatedQuery } from "test/utils"

describe("UpdateSubmissionMutation", () => {
  it("updates a submission and returns its new data payload", () => {
    const mutation = `
      mutation {
        updateConsignmentSubmission(
          input: { id: "108", artist_id: "andy-warhol", depth: "123", clientMutationId: "123123" }
        ) {
          clientMutationId
          submission {
            depth
          }
        }
      }
    `

    const rootValue = {
      submissionUpdateLoader: () =>
        Promise.resolve({
          id: "106",
          artist_id: "andy-warhol",
        }),
    }

    return runAuthenticatedQuery(mutation, rootValue).then(({ submissionUpdateLoader }) => {
      expect(submissionUpdateLoader).toMatchSnapshot()
    })
  })
})
