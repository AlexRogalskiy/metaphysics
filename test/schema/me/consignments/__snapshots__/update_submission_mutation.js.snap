// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateSubmissionMutation updates a submission and returns its new data payload 1`] = `undefined`;
